src/main/java/shef/data/ForYou.java

```
@@ -14,6 +14,31 @@
 
 package shef.data;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import javax.servlet.http.HttpServletRequest;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.PreparedQuery;
+
 public class ForYou implements RecipeFilter {
-  
-}
\ No newline at end of file
+
+  private DatastoreService datastore;
+
+  public ForYou(HttpServletRequest request) {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  public PreparedQuery getResults(Query query) {
+    return null;
```

It would be better to throw an UnsupportedOperationException or denote that method has not been implemented in this case: https://docs.oracle.com/javase/7/docs/api/java/lang/UnsupportedOperationException.html